src/patients/new/NewPatient.tsx

```diff
@@ -41,13 +45,39 @@ const NewPatient = () => {
   }
 
   const onSave = () => {
-    dispatch(createPatient(patient, onSuccessfulSave))
+    let isDuplicatePatient = false
+    const patientsObj = {}
+    Object.assign(patientsObj, patients)
+    Object.keys(patientsObj).forEach((patientInfo: any) => {
+      const loggedPatient = patients[patientInfo]
+      if (
+        loggedPatient.givenName === patient.givenName &&
+        loggedPatient.familyName === patient.familyName &&
+        loggedPatient.sex === patient.sex &&
+        loggedPatient.dateOfBirth === patient.dateOfBirth
+      ) {
+        setShowDuplicateNewPatientModal(true)
+        setDuplicatePatient(loggedPatient as Patient)
+        isDuplicatePatient = true
+      }
+    })
+    if (!isDuplicatePatient) {
+      dispatch(createPatient(patient, onSuccessfulSave))
+    }
   }
```

Comment:
After adding support for finding all possible duplicate patients, rather than just the first, this function would be a good candidate for `Arrays.filter` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter. 


```ts
const onSave = () => {
	 const duplicatePatients = patients.filter((existingPatient) => isPossibleDuplicatePatient(patient, existingPatient));
	
	if(duplicatePatients.length > 0) {
       setShowDuplicateNewPatientModal(true)
       setDuplicatePatient(existingPatient)
	} else {
      dispatch(createPatient(patient, onSuccessfulSave))
	}
}
```